source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -122,6 +138,91 @@ InstanceImpl::ThreadLocalPool::makeRequest(const std::string& key,
     client->redis_client_->addConnectionCallbacks(*client);
   }
 
+  // cache host for quick lookup via address for redirection support
+  auto host_cached_by_address = host_address_map_.find(host->address()->asString());
+  if (host_cached_by_address == host_address_map_.end()) {
+    host_address_map_[host->address()->asString()] = host;
+  }
+
+  return client->redis_client_->makeRequest(request, callbacks);
+}
+
+Common::Redis::Client::PoolRequest*
+InstanceImpl::ThreadLocalPool::makeRequestToHost(const std::string& host_address,
+                                                 const Common::Redis::RespValue& request,
+                                                 Common::Redis::Client::PoolCallbacks& callbacks) {
+  if (cluster_ == nullptr) {
+    ASSERT(client_map_.empty());
+    ASSERT(host_set_member_update_cb_handle_ == nullptr);
+    return nullptr;
+  }
+
+  auto colon_pos = host_address.rfind(":");
+  if ((colon_pos == std::string::npos) || (colon_pos == (host_address.size() - 1))) {
+    return nullptr;
+  }
+
+  std::string ip_address = host_address.substr(0, colon_pos);
+  bool ipv6 = (ip_address.find(":") != std::string::npos);
```

Comment:
nit: const (same elsewhere, try to const everything you can)

Reply:
ok